scripts/check-codegen.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Modification timestamp is the newest date between commit date and file modification
+function get_file_modification_timestamp() {
+	local pattern=$1
+	local fs_ts=$(for f in $(git status --porcelain "$pattern"|cut -c4-); do
+		if [[ -f "$f" ]]; then date -r "$f" '+%s'; else date '+%s'; fi # for deleted files, we use the current date
```

Comment:
d'ou sort le `"$f"`  ? c'est ce qui permet de faire le test d'archi mais je ne comprend pas d'ou il vient

Reply:
le `"$f"` vient du `for`, ce n'est pas le test d'archi mais le nom du fichier (issue de la commande `git status`)
on ne fait pas de `if` sur la date, le `if` qui est fait est pour détecter les fichiers supprimés (c'est le commentaire en fin de ligne)